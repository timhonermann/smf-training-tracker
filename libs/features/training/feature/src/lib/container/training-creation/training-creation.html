<stt-back-button [link]="backLink()" />
