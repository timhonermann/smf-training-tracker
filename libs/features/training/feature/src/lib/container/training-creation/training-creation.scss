:host {
  display: flex;
  flex-direction: column;
  gap: 16px;
  height: 100%;
}

.person-list {
  flex: 1;
  min-height: 0;
  overflow: auto;
}
