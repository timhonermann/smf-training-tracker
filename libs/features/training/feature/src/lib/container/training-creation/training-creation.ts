import { ChangeDetectionStrategy, Component, signal } from '@angular/core';
import { featureRoutes } from '@stt/shared/routing/model';
import { BackButton } from '@stt/shared/button/ui';

@Component({
  selector: 'stt-training-creation',
  imports: [BackButton],
  templateUrl: './training-creation.html',
  styleUrl: './training-creation.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TrainingCreation {
  readonly backLink = signal(`/${featureRoutes.TRAINING}`).asReadonly();
}
