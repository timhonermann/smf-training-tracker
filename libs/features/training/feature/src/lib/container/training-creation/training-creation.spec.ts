import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TrainingCreation } from './training-creation';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { expect } from 'vitest';
import { MockProvider } from 'ng-mocks';
import {
  TrainingLocationStore,
  TrainingStore,
} from '@stt/features/training/domain';

describe('TrainingCreation', () => {
  let component: TrainingCreation;
  let fixture: ComponentFixture<TrainingCreation>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [TrainingCreation],
      providers: [
        MockProvider(TrainingStore, {
          people: vi.fn(),
          selectedPeople: vi.fn(),
          create: vi.fn(),
          setSelectedPersonIds: vi.fn(),
        }),
        MockProvider(TrainingLocationStore, {
          entities: vi.fn(),
        }),
      ],
    })
      .overrideComponent(TrainingCreation, {
        set: { imports: [], schemas: [CUSTOM_ELEMENTS_SCHEMA] },
      })
      .compileComponents();

    fixture = TestBed.createComponent(TrainingCreation);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('backLink', () => {
    expect(component.backLink()).toMatchSnapshot();
  });
});
