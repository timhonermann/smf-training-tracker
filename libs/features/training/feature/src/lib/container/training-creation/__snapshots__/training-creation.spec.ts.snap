// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TrainingCreation > backLink 1`] = `"/training"`;
